### DISCUSSION TITLE
v3 unsupported in June 2023

### DISCUSSION THREAD

**Discussion by grepx** [braintree/braintree_android#719] [General]
Could somebody please clarify what will happen with v3 of the SDK in June? Will payments stop working or will it continue to work but without support for new features?

**Comment by sshropshire** [comment]
Hi @grepx payments will continue to work, but we'll stop providing updates to v3 of the Android SDK after June. Version 3 of the SDK depends on several deprecated Android methods.

Have you considered migrating to [Version 4](https://github.com/braintree/braintree_android/blob/master/v4.9.0%2B_MIGRATION_GUIDE.md) of the SDK?

**Reply by grepx** [reply]
Hi @sshropshire thanks for getting back - I am definitely planning to upgrade, just need to figure out how urgent the upgrade is. It sounds like the SDK will continue to work for a while longer than June - would it be reasonable to assume it will continue to work until the end of 2023?
